cmd/sandbox/start.go

```diff
@@ -164,34 +216,52 @@ func mountVolume(file, destination string) (*mount.Mount, error) {
 	return nil, nil
 }
 
-func downloadFlyteManifest(version string) error {
-	isGreater, err := util.IsVersionGreaterThan(version, FlyteMinimumVersionSupported)
+func monitorFlyteDeployment(ctx context.Context, appsClient v1.AppsV1Interface, nodeClient corev1k8s.NodeInterface) (int64, chan int64, chan error) {
+	countChan := make(chan int64)
+	chanErr := make(chan error)
+	var err error
+	defer func() {
+		if err != nil {
+			chanErr <- err
+		}
+	}()
+	total, err := k8s.GetFlyteDeploymentCount(ctx, appsClient)
 	if err != nil {
-		return err
-	}
-	if !isGreater {
-		return fmt.Errorf("version flag only support %s+ flyte version", FlyteMinimumVersionSupported)
+		return total, countChan, chanErr
 	}
-	response, err := util.GetRequest(GithubURL, fmt.Sprintf(FlyteReleaseURL, version))
+
+	countChan, chanErr = watchFlyteDeploymentStatus(ctx, appsClient, nodeClient)
+
+	return total, countChan, chanErr
+}
+
+func downloadFlyteManifest(version string) error {
+	release, err := util.CheckVersionExist(version, "flyte")
 	if err != nil {
 		return err
 	}
-	if err := util.WriteIntoFile(response, FlyteManifest); err != nil {
-		return err
+	var manifestURL = ""
+	for _, v := range release.Assets {
+		if v.GetName() == "flyte_sandbox_manifest.yaml" {
+			manifestURL = *v.BrowserDownloadURL
+		}
+	}
+	if len(manifestURL) > 0 {
+		response, err := http.Get(manifestURL)
+		if err != nil {
+			return err
+		}
+		defer response.Body.Close()
+		if response.StatusCode != 200 {
+			return fmt.Errorf("someting goes wrong while downloading the flyte release")
+		}
+		data, err := ioutil.ReadAll(response.Body)
+		if err != nil {
+			return err
+		}
+		if err := util.WriteIntoFile(data, FlyteManifest); err != nil {
+			return err
+		}
```

Comment:
Create this as part of a downloader utiity which can be reused at multiple places